src/core/toga/widgets/numberinput.py

```diff
@@ -126,7 +135,7 @@ def value(self, value):
             self._value = self.min_value
         elif self.max_value is not None and value > self.max_value:
             self._value = self.max_value
-        self._impl.set_value(value)
+        self._impl.set_value(self._value)
```

Comment:
This line was correct - the value that is pass down as an argument should be the "as provided" value; if the implementation wants the "parsed and processed" value, it can interrogate `self._impl.value`.